bcda/auth/middleware.go

```diff
@@ -116,6 +120,27 @@ func RequireTokenAuth(next http.Handler) http.Handler {
 	})
 }
 
+// CheckBlacklist checks the auth data is associated with a blacklisted entity
+func CheckBlacklist(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ad, ok := r.Context().Value(AuthDataContextKey).(AuthData)
+		if !ok {
+			log.Error()
+			oo := responseutils.CreateOpOutcome(responseutils.Error, responseutils.Exception, responseutils.Not_found, "AuthData not found")
+			responseutils.WriteError(oo, w, http.StatusNotFound)
+			return
+		}
+
+		if ad.Blacklisted {
+			oo := responseutils.CreateOpOutcome(responseutils.Error, responseutils.Exception, responseutils.RequestErr,
+				fmt.Sprintf("ACO (CMS_ID: %s) is blacklisted", ad.CMSID))
```

Comment:
Do we want to use this term ("blacklisted") in a public facing error message?  Maybe we should just say something about being unauthorized?